### PR TITLE
Implement My Top Posts View

### PR DESCRIPTION
**Pull request by judyliu14**
- added my top posts view
- fixed refresh error between different tabs that occurs when starring a post or creating a new post

### CODE REVIEW

File: database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Models/PostViewModel.swift
Comment (judyliu14) [reply database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Models/PostViewModel.swift:36]: On the new commit I have removed the handler with .onDisappear but am not sure if I have implemented correctly.

File: database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Views/MyPostsView.swift
Comment (maksymmalyhin) [inline database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Views/MyPostsView.swift:35]: Just a small correction. I don't think the view needs to know about ref handles as it is an implementation detail. The view should just forward the events important for the View Model. For this case it basically means renaming the method `UserViewModel.removeRefHandles()` to `UserViewModel.onViewDisappear()`.  The same applicable for `onAppear` here and for other views below.

File: database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Views/TopPostsView.swift
Comment (maksymmalyhin) [inline database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Views/TopPostsView.swift:20]: How much different are `RecentPostsView` and `TopPostsView`? Can we have a generic `PostsView` that accepts a parameter like `type` that can be used instead of `tabOpened`?

**Review by maksymmalyhin — APPROVED** [review]
Mostly LGTM, a couple minor comments.

File: database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Views/TopPostsView.swift
Comment (judyliu14) [reply database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Views/TopPostsView.swift:20]: Do you mean renaming tabOpened to type or something else? RecentPostView and TopPostsView are retrieving data from two different parts of Firebase Realtime Database.

Comment (maksymmalyhin) [reply database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Views/TopPostsView.swift:20]: My main question is if `RecentPostsView` and `TopPostsView` display a given list of posts anyhow different? It looks to me that no and the only difference is `tabOpened` value. If it is so, I don't think we need to have two different classes, we can just have a single one and initialize it with `tabOpened` value.

My second point is about `tabOpened` name itself. It seems not exactly precise to me. If we have a view for a list of posts I would name the parameter like `postType` or something along these lines. I'm not too strong about it though.

Comment (judyliu14) [reply database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Views/TopPostsView.swift:20]: I see did you mean combining all 3 `RecentPostsView`, `MyPostsView`, and `TopPostsView` into one file and pass in the tabOpened value? Currently in each view it contains its own `@StateObject` of type `UserViewModel` that refreshes on appear, it seems like the app is still functional after combining the 3 views into one file. So to clarify I will pass in the `tabOpened` and `navigation title` to a single class instead of having different classes?

Comment (maksymmalyhin) [reply database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Views/TopPostsView.swift:20]: SGTM

File: database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Models/UserViewModel.swift
Comment (paulb777) [inline database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Models/UserViewModel.swift:140]: Can tabOpened be an enum instead of a String to avoid all the `rawValue`s?

File: database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Models/PostViewModel.swift
Comment (morganchen12) [reply database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Models/PostViewModel.swift:36]: Looks fine to me 👍

**Review by morganchen12 — APPROVED** [review]
LGTM, but please wait for all other approvers before merging.

File: database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Views/TopPostsView.swift
Comment (judyliu14) [reply database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Views/TopPostsView.swift:20]: Sounds good. Will restructure on the next PR.

**Review by paulb777 — APPROVED** [review]

**Pull request merged**
